command/plugins_required.go

```diff
@@ -0,0 +1,109 @@
+package command
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"runtime"
+	"strings"
+
+	pluginsdk "github.com/hashicorp/packer-plugin-sdk/plugin"
+	plugingetter "github.com/hashicorp/packer/packer/plugin-getter"
+	"github.com/mitchellh/cli"
+)
+
+type PluginsRequiredCommand struct {
+	Meta
+}
+
+func (c *PluginsRequiredCommand) Synopsis() string {
+	return "List plugins required by a config"
+}
+
+func (c *PluginsRequiredCommand) Help() string {
+	helpText := `
+Usage: packer plugins required <path>
+
+  This command will list every Packer plugin required by a Packer config.
```

Comment:
I'm trying to think how to word this in the help message. But the messaging should indicate that the command will read the `required_plugins` block in a Packer config and list every required plugin. 

It is confusing to run it against a valid .pkr.hcl file that does not have a required_plugins block and see no output. Maybe an alternative here could be to error and so the Packer config does not contain a required_plugins block.

Reply:
Good point ! Ehm, after some thinking, we could say:

```
  This command will list every Packer plugin required by a Packer config in
  packer.required_plugins blocks. All binaries matching the required version
  constrain and the current OS and Architecture will be listed.
```

Reply:
I think we should not error here, but maybe display a warning ?

Reply:
The updated messaging looks great!

Reply:
A warning sounds good.

Reply:
I added a warning ! It goes like: 

```
No plugins requirement found, make sure you reference a Packer config
containing a packer.required_plugins block. See
https://www.packer.io/docs/templates/hcl_templates/blocks/packer#specifying-plugin-requirements
for more info.
```